frame/staking/src/lib.rs

```diff
@@ -745,17 +745,46 @@ enum Releases {
 	V4_0_0,
 	V5_0_0, // blockable validators.
 	V6_0_0, // removal of all storage associated with offchain phragmen.
+	V7_0_0, // keep track of number of nominators / validators in map
 }
 
 impl Default for Releases {
 	fn default() -> Self {
-		Releases::V6_0_0
+		Releases::V7_0_0
 	}
 }
 
 pub mod migrations {
 	use super::*;
 
+	pub mod v7 {
+		use super::*;
+
+		pub fn pre_migrate<T: Config>() -> Result<(), &'static str> {
+			assert!(CurrentValidatorsCount::<T>::get().is_zero(), "CurrentValidatorsCount already set.");
+			assert!(CurrentNominatorsCount::<T>::get().is_zero(), "CurrentNominatorsCount already set.");
```

Comment:
Since `Current{Nominators, Validator}Count` is only introduced in this PR, is it safe to assume the runtime upgrade process sets `:code` and then applies the `on_runtime_upgrade` hooks?

Reply:
Note that if the value does not exist, then the storage APIs always return `Default`, which is `0` for numbers. So being zero and non existent is the same. (And if you happen to not like this then you can use `Option<_>` for it)

Reply:
Valuequery will return default. Optionquery will return none

Reply:
danke